### PR TITLE
fix types

### PR DESCRIPTION
**Pull request by MaartenFaddegon**
This fixes a few minor type errors that propagated to projects depending on `ExAzureKeyVault`.

### CODE REVIEW

**Comment by RamonPage** [comment]
Hi @MaartenFaddegon, thank you for your contribution!

Could you tell me more about your use case? It looks like you are changing more than just types and this is failing some tests.

**Comment by MaartenFaddegon** [comment]
I'll update the pull request to be of just the updated specs. 

Those code changes were the result of me trying to figure out what was wrong and I should have not included them. I hadn't expected them to affect any unit tests though ...

**Comment by MaartenFaddegon** [comment]
After you merged this and Anders' pull request, could you maybe make a new release on Hex?

File: mix.exs [hunk 1]
```diff
@@ -56,6 +56,7 @@ defmodule ExAzureKeyVault.MixProject do
     [
       {:httpoison, "~> 1.2.0"},
       {:poison, "~> 4.0.0"},
+      {:dialyxir, "~> 0.5", only: [:dev], runtime: false},
```
Comment (RamonPage) [inline mix.exs]: Could you remove this change from this PR as well? I would like to keep this project as lean as possible.

Please create an issue or a separate PR if you want to discuss more about this dependency.

**Comment by RamonPage** [comment]
Of course!

File: mix.exs (hunk 1 above)
Comment (MaartenFaddegon) [reply mix.exs]: I will remove the Dialyzer dependency, but I highly recommend you look into Dialyzer: it allows you to run typechecks and could have prevented the mistakes fixed in this PR.

Comment (RamonPage) [reply mix.exs]: I really appreciate the suggestion. I'll take that into consideration later.

**Pull request merged**